src/data/encode.ts

```diff
@@ -120,3 +121,190 @@ export const decode = async (data: EncodedData): Promise<string> => {
 
   return decoded;
 };
+
+// -----------------------------------------------------------------------------
+// binary encoding
+// -----------------------------------------------------------------------------
+
+type FileEncodingInfo = {
+  version: 1;
+  compression: "pako@1";
+  encryption: "AES-GCM" | null;
+};
+
+// -----------------------------------------------------------------------------
+/** how many bytes we use to encode how many bytes the next chunk has.
+ * Corresponds to DataView setter methods (setUint32, setUint16, etc).
+ *
+ * NOTE ! values must not be changed, which would be backwards incompat change !
+ */
+const NEXT_CHUNK_SIZE_DATAVIEW_BYTES = 4;
+const CHUNKS_COUNT_DATAVIEW_BYTES = 1;
+// -----------------------------------------------------------------------------
+
+/**
```

Comment:
Can you put a description of the file format somewhere as a comment? Something like this as a reference.

![image](https://user-images.githubusercontent.com/197597/135914284-0d5f0dd8-8f6b-484a-94b1-d7f08a1a2e9f.png)

Reply:
sounds good